Added simulator bridge that publishes sub pos, orientation, and depth

dcrossman opened the pull request.

dcrossman committed:
Added simulator bridge that publishes sub pos, orientation, and depth

skallaher commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,92 @@
+{
+    geometry_msgs::Vector3 position_msg;
+    geometry_msgs::Quaternion orientation_msg;
+    geometry_msgs::Vector3 lin_accel_msg;
```
You commented out the publisher for the `lin_accel` shouldn't this also be commented out?

dcrossman committed:
Quick fix to sim bridge

dcrossman committed:
Simulator bridge now publishes positions of pool obstacles.
It publishes an ObstaclePosArray msg.
Names of obstacles to find are listed in param/obstacles.yaml.

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+	ros::Subscriber orient_sub = nh.subscribe("gazebo/model_states", 1, modelStatesCallback);
+
+    int rate;
+    if(!nh.getParam("control/rate", rate))
```
Add brackers

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+    // Info not being published right now
+    //lin_accel_pub = nh.advertise<geometry_msgs::Vector3>("lin_accel", 1);
+
+	ros::Subscriber orient_sub = nh.subscribe("gazebo/model_states", 1, modelStatesCallback);
```
unindent

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+    }
+
+    // I use spinOnce and sleeps here because the simulator publishes
+    // at a very high rate and we don't need every message
```
Period

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+
+    // I use spinOnce and sleeps here because the simulator publishes
+    // at a very high rate and we don't need every message
+    // You can throttle subscriber input in other ways but
```
comma

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+    // I use spinOnce and sleeps here because the simulator publishes
+    // at a very high rate and we don't need every message
+    // You can throttle subscriber input in other ways but
+    // doesn't require extra packages or anything
```
Period

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+ros::Publisher orientation_pub;
+ros::Publisher depth_pub;
+ros::Publisher obstacle_pos_pub;
+//ros::Publisher lin_accel_pub;
```
Remove commented out lines of code  if you plan on adding them in future pull requests.

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+{
+    geometry_msgs::Vector3 position_msg;
+    geometry_msgs::Quaternion orientation_msg;
+    //geometry_msgs::Vector3 lin_accel_msg;
```
Remove

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+        // not found in array.
+        if(it != msg.name.end())
+        {
+            // Calculate index
```
Calculate index in array from the iterator element.

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+        if(it != msg.name.end())
+        {
+            // Calculate index
+            auto idx = std::distance(msg.name.begin(), it);
```
Don't make this auto for clarity.

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+
+    obstacle_pos_pub.publish(object_array);
+
+    //lin_accel_pub.publish(lin_accel_msg);
```
Remove commented out line and newline above.

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+
+int main(int argc, char **argv)
+{
+	ros::init(argc, argv, "simulator_bridge");
```
Fix indentation.

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+    position_pub = nh.advertise<geometry_msgs::Vector3>("position", 1);
+    orientation_pub = nh.advertise<geometry_msgs::Quaternion>("orientation", 1);
+    depth_pub = nh.advertise<std_msgs::Float32>("depth", 1);
+    obstacle_pos_pub = nh.advertise<robosub::ObstaclePosArray>("obstacle_positions", 1);
```
change topic to obstacles/positions

ryan-summers commented on src/sensors/simulator_bridge.cpp:
```diff
@@ -0,0 +1,131 @@
+    depth_pub = nh.advertise<std_msgs::Float32>("depth", 1);
+    obstacle_pos_pub = nh.advertise<robosub::ObstaclePosArray>("obstacle_positions", 1);
+    // Info not being published right now
+    //lin_accel_pub = nh.advertise<geometry_msgs::Vector3>("lin_accel", 1);
```
Remove commented out line of code.

ryan-summers reviewed (changes requested):
Remove commented lines of code and all unneeded newlines.

dcrossman committed:
Post review simulator bridge changes.

ryan-summers reviewed (approved):
Everything is looking good from here pending addition of the gazebo packages to the robosub packages.

dcrossman committed:
Updated travis
